paymentsheet/src/androidTest/java/com/stripe/android/paymentsheet/LinkTest.kt

```diff
@@ -93,42 +98,135 @@ internal class LinkTest {
+        integrationType = integrationType,
+        paymentOptionCallback = { paymentOption ->
+            assertThat(paymentOption?.label).endsWith("4242")
+        },
+        resultCallback = ::assertCompleted,
+    ) { testContext ->
+        networkRule.enqueue(
+            host("api.stripe.com"),
+            method("GET"),
+            path("/v1/elements/sessions"),
+        ) { response ->
+            response.testBodyFromFile("elements-sessions-requires_pm_with_link_ps_mode.json")
+        }
+
+        when (testContext) {
+            is LinkTestRunnerContext.PaymentSheet -> {
+                testContext.context.presentPaymentSheet {
+                    presentWithPaymentIntent(
+                        paymentIntentClientSecret = "pi_example_secret_example",
+                        configuration = emptyConfiguration(),
+                    )
+                }
+            }
+            is LinkTestRunnerContext.FlowController -> {
+                testContext.context.configureFlowController {
+                    configureWithPaymentIntent(
+                        paymentIntentClientSecret = "pi_example_secret_example",
+                        configuration = emptyConfiguration(),
+                        callback = { success, error ->
+                            assertThat(success).isTrue()
+                            assertThat(error).isNull()
+                            presentPaymentOptions()
+                        },
+                    )
+                }
+            }
+        }
+
+        page.fillOutCardDetails()
+
+        networkRule.enqueue(
+            method("POST"),
+            path("/v1/consumers/sessions/lookup"),
+        ) { response ->
+            response.testBodyFromFile("consumer-session-lookup-success.json")
+        }
+
+        page.clickOnLinkCheckbox()
+        page.fillOutLinkEmail()
+        page.fillOutLinkPhone()
+
+        Espresso.closeSoftKeyboard()
+
+        networkRule.enqueue(
+            method("POST"),
+            path("/v1/consumers/accounts/sign_up"),
+        ) { response ->
+            response.testBodyFromFile("consumer-accounts-signup-success.json")
+        }
+
+        networkRule.enqueue(
+            method("POST"),
+            path("/v1/consumers/payment_details"),
+            /*
+             * Ensures we are passing the full expiration year and not the
+             * 2-digit shorthand (should send "2034", not "34")
+             */
+            bodyPart("card%5Bexp_year%5D", "2034"),
+            bodyPart("active", "true")
+        ) { response ->
+            response.testBodyFromFile("consumer-payment-details-success.json")
+        }
+
+        networkRule.enqueue(
+            method("POST"),
+            path("/v1/consumers/payment_details/share"),
+        ) { response ->
+            response.testBodyFromFile("consumer-payment-details-success.json")
+        }
+
+        networkRule.enqueue(
```

Comment:
We should add an assert that the correct passthrough mode request params are sent to confirm.